src/chapters/unit-testing/testing-in-csharp.rst

```diff
@@ -0,0 +1,115 @@
+Testing in C#
+=============
+
+.. index:: ! automated testing, ! unit testing, ! MSTest
+
+You already recognize the importance of testing your code. 
+**Automated testing** in C# serves the same purpose as it does in any language: 
+to ensure that your code works as expected every time it runs. Tests also 
+function as documentation, giving a fellow programmer instructions 
+on how to properly execute your classes and methods.
+
+This course covers **unit testing** in C# with a test framework called
+**MSTest**. Unit testing breaks down the codebase into its smallest
+building blocks, which are individual statements and methods. 
+
+.. tip::
+
+   If you're not yet familiar with the benefits of automated testing,
+   you can check out
+   `this section <https://education.launchcode.org/intro-to-professional-web-dev/chapters/unit-testing/why-test.html>`__
+   in another LaunchCode text.
+
+Why We Test
+-----------
+
+.. index:: ! refactor, ! test-driven development, ! TDD
+
+Refactoring
+^^^^^^^^^^^
+
+When we **refactor** code, we rewrite without adding new features. Refactoring can 
+increase efficiency at runtime, but it may also mean inadvertently introducing some bugs in the process.
+Unit tests verify the most basic functionality of your code, thus safeguarding against 
+bugs introduced in refactoring. 
+
+Imagine this common workflow: 
+
+#. You practice **test-driven development (TDD)**, writing your tests to stipulate your class's code. 
+
+#. You write your class's code to pass they tests. 
+
+#. Later, a stakeholder in the project requests that you refactor your code using 
+   different syntax.
+
+The features will be the same, but the implementation changes. 
+Because the changes in implementation do not require changes to the outcomes, 
+unit tests will help in this scenario. If your tests continue
+to pass after the refactor, you can move on, knowing you have not 
+inadvertently introduced a bug. Writing tests just once provides innumerable 
+benefits for the whole lifetime of the codebase.
+
+Documentation
+^^^^^^^^^^^^^
+
+Unit tests serve as key documentation for fellow programmers. Again, 
+because they address the most fundamental tasks of your classes,
+unit tests serve as live-code use-cases. You may also have an 
+external documentation directory with examples of how to run your
+project, or perhaps you have been writing comments within your code
+to best communicate with your teammates about your changes. Both of
+these are great choices and should be done when possible. However, they 
+also require more forethought to maintain. Each time you update
+your code, you might not remember to update the documentation and 
+comments. With unit testing, however, you have a more obvious reminder
+that a change has been made if a test fails.
```

Comment:
"With unit testing, you have an obvious reminder that you made changes when a previously-passing test fails."